PyInstaller/building/build_main.py

```diff
@@ -426,17 +440,26 @@ def assemble(self):
                 if node_type not in VALID_MODULE_TYPES:
                     continue
 
-                # Run all post-graph hooks for this module.
-                for hook_file in hooks_cache[imported_name]:
-                    # Import hook module from a file.
-                    imphook_object = ImportHook(imported_name, hook_file)
-                    # Expand module dependency graph.
-                    imphook_object.update_dependencies(self.graph)
-                    # Update cache of binaries and datas.
-                    additional_files_cache.add(imported_name, imphook_object.binaries, imphook_object.datas)
-
+                # Run hooks for all parent packages but only if they were not
+                # applied yet.
+                #   'aaa.bb.c.dddd' ->  ['aaa', 'aaa.bb', 'aaa.bb.c']
+                parent_pkgs = module_parent_packages(imported_name)
+                if parent_pkgs:  # 'imported_name' is not top-level module.
```

Comment:
This if is useless, he for-loop does the same - it only requires `module_parent_packages()` to always return a list.
The variable `parent_pkgs` can then be saved, too.